### DISCUSSION TITLE
Phyton in too large to convert to C long

### DISCUSSION THREAD

**Discussion by Rerol87gt** [nagadomi/nunif#105] [General]
Hey, have a good weekend, and I want to say this tool is amazing. Thank you for sharing. and I exported a video for some depthmaps, which I had already edited. I would like to re-import, and I select yml, but when I choose convert, it shows Phyton as too large to convert to C. Is there anything I am doing wrong?

![Screenshot 2024-03-30 144850](https://github.com/nagadomi/nunif/assets/165326995/cfe143a8-b49b-45f4-995f-3bd713e3f795)

**Comment by nagadomi** [comment]
That error is caused by floating point FPS.
What is the fps value defined in yml? (When yml is specified as input, fps defined in yml is used as fps for the output video)
Currently, only the following FPS are specially handled.

```
29.97 = 30000 / 1001
23.976 = 24000 / 1001
59.94 = 60000 / 1001
```

If a floating point FPS value cannot be represented by a fraction in 32bit int range, an error occurs.
For example, `0.25=1/4` is fine, but `0.124=553942754166571 / 4503599627370496` is an error.

**Reply by nagadomi** [reply]
Also, this is probably a bug in the export feature, so if you did not enter the fps value yourself, it can be fixed in the future.
However, it is not possible to fix an existing yml.

**Comment by nagadomi** [comment]
Fixed. Run `update.bat`  again.
However, existing yml file will not be corrected.
If you do not want to re-export, you must manually correct the fps values. 
For example,
```
29.97002997002997 → 30000/1001
23.976023976023978 → 24000/1001
59.94005994005994 → 60000/1001
```
If you have any questions, please let me know.
